homeassistant/components/swiss_public_transport/config_flow.py

```diff
@@ -63,25 +94,97 @@ async def async_step_user(
                     user_input[CONF_DESTINATION],
                     session,
                     via=user_input.get(CONF_VIA),
+                    time=user_input.get(CONF_TIME),
                 )
-                try:
-                    await opendata.async_get_data()
-                except OpendataTransportConnectionError:
-                    errors["base"] = "cannot_connect"
-                except OpendataTransportError:
-                    errors["base"] = "bad_config"
-                except Exception:  # pylint: disable=broad-except
-                    _LOGGER.exception("Unknown error")
-                    errors["base"] = "unknown"
+                err = await self.fetch_connections(opendata)
+                if err:
+                    errors["base"] = err
                 else:
-                    return self.async_create_entry(
-                        title=unique_id,
-                        data=user_input,
+                    if user_input[CONF_TIME_MODE] == "now":
+                        unique_id = unique_id_from_config(user_input)
+                        await self.async_set_unique_id(unique_id)
+                        self._abort_if_unique_id_configured()
+                        return self.async_create_entry(
+                            title=unique_id,
+                            data=user_input,
+                        )
+                    self.user_input = user_input
+                    return self.async_show_form(
```

Comment:
You need to save the input data and load the next step, not show a form for a step which you are not in.